src/vs/workbench/services/editor/browser/editorOverrideService.ts

```diff
@@ -625,6 +629,15 @@ export class EditorOverrideService extends Disposable implements IEditorOverride
 		}
 		return false;
 	}
+
+	private isDiffEditorInput(editor: UntypedEditorInput): editor is IResourceDiffEditorInput {
+		const testInput = editor as IResourceDiffEditorInput;
+		return testInput.originalInput !== undefined && testInput.modifiedInput !== undefined;
+	}
+
+	private getEditorResource(editor: UntypedEditorInput): URI | undefined {
```

Comment:
Should this be added to `EditorResourceAccessor` as an overload?

Reply:
I've enriched the `EditorResourceAccessor` to handle this case.